packages/realm/src/Realm.ts

```diff
@@ -107,20 +165,54 @@ type InternalConfig = {
 };
 
 export class Realm {
-  public static Object = RealmObject;
+  /** @deprecated Please use named imports */
+  public static App = App;
+  /** @deprecated Please use named imports */
+  public static Auth = { EmailPasswordAuth, ApiKeyAuth };
+  /** @deprecated Please use named imports */
+  public static BSON = BSON;
+  /** @deprecated Please use named imports */
+  public static ClientResetMode = ClientResetMode;
+  /** @deprecated Please use named imports */
   public static Collection = Collection;
+  /** @deprecated Please use named imports */
+  public static ConnectionState = ConnectionState;
+  /** @deprecated Please use named imports */
+  public static Dictionary = Dictionary;
+  /** @deprecated Please use named imports */
+  public static List = List;
+  // TODO: Decide if we want to deprecate this as well
+  public static Object = RealmObject;
```

Comment:
Only that `Object` is a class provided by the runtime globally and `import { Object } from "realm"` would shadow that (and confuse some static analysis tools, like linters, etc.). In a file where an end-user need access to the `Object` provided by the runtime and that provided by us, they could rename on import:

```
import { Object as RealmObject } from "realm";
// Use RealmObject
```

Or alternatively import us as a namespace (which won't be deprecated):

```
import * as Realm from `realm`;
// Use Realm.Object
```

The latter could be a bit confusing since `Realm` in that case wouldn't be a constructor.

By not deprecating on the line we're discussing, we would allow:

```
import { Realm } from "realm";
// Use Realm.Object
// Realm is still a constructor
```

I suspect this is the reason the namespaced API got introduced in the first case.

Reply:
Gotcha, thanks for the explanation. By the way, for named exports, partly due to want you just mentioned with having to rename the import, an alternative would be to call our realm objects Realm<Name> (e.g. RealmDictionary, RealmObject, etc.). I've seen other SDKs take this approach as well.

Reply:
> could we not export it as `RealmObject`

We could .. We should probably think long an hard about that tough. The package name ("realm") is already a namespace (not the TS kind of namespace ...), so in general I tend to opt towards not include a name twice (both in the thing and the thing / namespace that contains it)